LoopOnboardingKitUI/View Controllers/OnboardingUICoordinator.swift

```diff
@@ -0,0 +1,425 @@
+    }
+
+    private func stepFinished() {
+        if let nextScreen = currentScreen.next() {
+            navigate(to: nextScreen)
+        } else {
+            completionDelegate?.completionNotifyingDidComplete(self)
+        }
+    }
+
+    private func navigate(to screen: OnboardingScreen) {
+        var viewControllers = self.viewControllers
+
+        // Remove the Nightscout chooser from the view controller hierarchy if the Nightscout service is fully onboarded
+        if currentScreen == .nightscoutChooser && service?.isOnboarded == true {
+            screenStack.removeLast()
+            viewControllers.removeLast()
+        }
+
+        screenStack.append(screen)
+        viewControllers.append(viewControllerForScreen(screen))
+        setViewControllers(viewControllers, animated: true)
+    }
+
+    private func setupWithNightscout() {
+        if let service = service {
+            if service.isOnboarded {
+                stepFinished()
+            } else if let serviceUI = service as? ServiceUI {
+                var settingsViewController = serviceUI.settingsViewController(colorPalette: colorPalette)
+                settingsViewController.serviceOnboardDelegate = self
+                settingsViewController.completionDelegate = self
+                show(settingsViewController, sender: self)
+            } else {
+                fatalError("Failure to setup service (without UI) with identifier: \(service.serviceIdentifier)")
+            }
+        } else {
+            switch serviceProvider.setupService(withIdentifier: OnboardingUICoordinator.serviceIdentifier) {
+            case .failure(let error):
+                log.debug("Failure to create and setup service with identifier '%{public}@': %{public}@", OnboardingUICoordinator.serviceIdentifier, String(describing: error))
+            case .success(let success):
+                switch success {
+                case .userInteractionRequired(var setupViewController):
+                    setupViewController.serviceCreateDelegate = self
+                    setupViewController.serviceOnboardDelegate = self
+                    setupViewController.completionDelegate = self
+                    show(setupViewController, sender: self)
+                case .createdAndOnboarded(let service):
+                    serviceCreateNotifying(didCreateService: service)
+                    serviceOnboardNotifying(didOnboardService: service)
+                    stepFinished()
+                }
+            }
+        }
+    }
+
+    private func setupWithoutNightscout() {
+        stepFinished()
+    }
+
+    private func constructTherapySettingsViewModel(therapySettings: TherapySettings) -> TherapySettingsViewModel? {
+        let supportedBasalRates = (1...600).map { round(Double($0) / Double(1.0/0.05) * 100.0) / 100.0 }
+
+        let maximumBasalScheduleEntryCount = 24
+
+        let supportedBolusVolumes = (1...600).map { Double($0) / Double(1/0.05) }
```

Comment:
For both DIY and Tidepool; this should change to get actual volumes/rates from the selected pump. Do we have a ticket to track the Tidepool side of this?

Reply:
I cannot immediately find an associated Jira ticket, so created one. https://tidepool.atlassian.net/browse/LOOP-3112